packages/vite/src/node/plugins/asset.ts

```diff
@@ -146,11 +146,11 @@ export function checkPublicFile(
   }
 }
 
-export function fileToUrl(
+export async function fileToUrl(
   id: string,
   config: ResolvedConfig,
   ctx: PluginContext
-): string | Promise<string> {
+): Promise<string> {
```

Comment:
Had to change the return type to `Promise<string>` for the `.catch` syntax to work. And for TypeScript to not complain about `Promise<string>`, I had to add the `async` keyword.

I think it makes sense to return a promise since either way to consumer would always have to await.